core/domain/skill_services.py

```diff
@@ -1065,6 +1065,36 @@ def skill_has_associated_questions(skill_id):
     return len(question_ids) > 0
 
 
+def filtered_skills_with_none_type_keys(degrees_of_mastery):
```

Comment:
Please add tests for this function.

Reply:
`test_filter_skills_by_mastery` is fully dependent on this function. should I create a separate test?

Reply:
yes, please